src/main/java/problems/impl/TransactionalStack.java

```diff
@@ -0,0 +1,115 @@
+package problems.impl;
+
+import problems.exceptions.TransactionRolledbackException;
+
+import java.util.EmptyStackException;
+import java.util.Stack;
+
+public class TransactionalStack<T> {
+    private final Stack<T> stack = new Stack<>();
+    private final Stack<TransactionEvent<T>> transactionEvents = new Stack<>();
+    private Stack<TransactionEvent<T>> rolledbackTransactionEvents = new Stack<>();
```

Comment:
Because I'm using the `public` [`push` / `pop` methods in the implementation of the `rollback` method](https://github.com/jaebradley/programmingProblems/pull/88/files#diff-e3d1c6c4659eaf77c2f0884d468e1fa4R105) I need to check whether I'm in the middle of a rollback - else, I might not get to the same state pre-transaction.